kerckhoff/packages/serializers.py

```diff
@@ -172,3 +174,82 @@ def get_version_data(self, obj: Package):
     class Meta(PackageSerializer.Meta):
         fields = PackageSerializer.Meta.fields + ("version_data",)
         read_only_fields = PackageSerializer.Meta.read_only_fields + ("version_data",)
+
+class PackageInfoSerializer(serializers.ModelSerializer):
+    package_set = serializers.StringRelatedField()
+    latest_version = serializers.StringRelatedField()
+    description = serializers.SerializerMethodField()
+    folder_id = serializers.SerializerMethodField()
+    folder_url = serializers.SerializerMethodField()
+    metadata = serializers.SerializerMethodField()
+    data = serializers.SerializerMethodField()
+    cached_article_preview = serializers.SerializerMethodField()
+
+    def get_description(self, obj: Package):
+        obj.fetch_cache()
+        versionSerializer = PackageVersionSerializer(obj.get_version(obj.latest_version.id_num))
+        return versionSerializer.data["version_description"]
+
+    def get_folder_id(self, obj: Package):
+      return obj.metadata["google_drive"]["folder_id"]
+    
+    def get_folder_url(self, obj: Package):
+      return obj.metadata["google_drive"]["folder_url"]
+
+    def get_metadata(self, obj: Package):
+      return {}
+    
+    def get_data(self, obj: Package):
+        obj.fetch_cache()
+        package_items = obj.get_version(obj.latest_version.id_num).packageitem_set.all() 
```

Comment:
Ok when I wrote this line a few years ago I didn't realize that obj.latest_version.packageitem_set.all() is good enough.

Reply:
For this one, again please add handling for the case where no versions have been created yet

Reply:
Done.